cloud/defaults.go

```diff
@@ -100,6 +102,19 @@ func GenerateDataDiskName(machineName, nameSuffix string) string {
 	return fmt.Sprintf("%s_%s", machineName, nameSuffix)
 }
 
+// GenerateRoleAssignmentName generates a valid GUID based on an input string.
+func GenerateRoleAssignmentName(s string) string {
+	h := md5.New() // #nosec
```

Comment:
I'm not sure if this is acceptable, the issue is that the other options for hash aren't 16 bytes long so I wouldn't be able to convert them to UUID as-is. This isn't really "insecure" because this data isn't a secret but it's getting flagged by gosec. Open to suggestions for not using md5 if anyone has any.

@nader-ziada @devigned

Reply:
md5 is considered unsecure because it is easy to reverse, but I don't think this has any of the information used here (subscription, location, machine name) is confidential

Reply:
can you add a comment so its clear why we have the `#nosec`

Reply:
+1 to `// #nosec` with comment.

I don't see a security issue here since it is scoped to within the sub and as @nader-ziada said, we are using common input to derive the hash.

There is a possibility that for some input we map to some same output, which would cause a hash collision. One can generate a hash collision in seconds using MD5.

What do you think about just generating a UUID and storing it in the status?

Reply:
> What about adding the role assignment name to the spec and generate a UUID in a web hook?

That's the other option. I thought not adding to spec would be slightly better because otherwise users could potentially mess with it and it's not really an option we want exposed to the user. But given how complicated using a hash is it might make more sense, I'm not sure...